agreement_rebate/models/agreement_rebate_settlement.py

```diff
@@ -0,0 +1,243 @@
+        target_domains = self.line_ids.mapped('target_domain')
+        domain = expression.OR([safe_eval(d) for d in set(target_domains)])
+        # if self.rebate_type == 'line' and len(self.line_ids) > 1:
+        #     domain = expression.OR([safe_eval(
+        #         l.target_domain) for l in self.line_ids])
+        # else:
+        #     domain = safe_eval(self.line_ids[:1].target_domain)
+        return {
+            'name': _('Details'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'account.invoice.report',
+            'view_mode': 'pivot,tree',
+            'domain': domain,
+            'context': self.env.context,
+        }
+
+    def action_show_settlement(self):
+        action = self.env.ref(
+            'agreement_rebate.agreement_rebate_settlement_action').read()[0]
+        if len(self) == 1:
+            form = self.env.ref(
+                'agreement_rebate.agreement_rebate_settlement_form')
+            action['views'] = [(form.id, 'form')]
+            action['res_id'] = self.id
+        else:
+            action['domain'] = [('id', 'in', self.ids)]
+        return action
+
+    def action_show_agreement(self):
+        agreements = self.line_ids.mapped('agreement_id')
+        action = self.env.ref('agreement.agreement_action').read()[0]
+        if len(agreements) == 1:
+            form = self.env.ref('agreement.agreement_form')
+            action['views'] = [(form.id, 'form')]
+            action['res_id'] = agreements.id
+        else:
+            action['domain'] = [('id', 'in', agreements.ids)]
+        return action
+
+
+class AgreementRebateSettlementLine(models.Model):
+    _name = 'agreement.rebate.settlement.line'
+    _description = 'Agreement Rebate Settlement Lines'
+    _order = 'date DESC'
+
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company',
+        related='rebate_settlement_id.company_id',
+    )
+    rebate_settlement_id = fields.Many2one(
+        comodel_name='agreement.rebate.settlement',
+        string='Rebate settlement',
+        ondelete='cascade',
+    )
+    date = fields.Date(
+        related='rebate_settlement_id.date',
+        store=True,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Partner',
+    )
+    rebate_line_id = fields.Many2one(
+        comodel_name='agreement.rebate.line',
+        string='Rebate Line',
+    )
+    rebate_section_id = fields.Many2one(
+        comodel_name='agreement.rebate.section',
+        string='Rebate section',
+    )
+    target_domain = fields.Char()
+    amount_from = fields.Float(string="From", readonly=True)
+    amount_to = fields.Float(string="To", readonly=True)
+    percent = fields.Float(string="Percent", readonly=True)
+    amount_gross = fields.Float(string='Amount gross')
+    amount_invoiced = fields.Float(string='Amount invoiced')
+    amount_rebate = fields.Float(string='Amount rebate')
```

Comment:
```suggestion
    total_amount = fields.Float(string='Amount rebate')
```
Same question about digits and currencies